src/utils/rest-api.ts

```diff
@@ -0,0 +1,296 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { APP_NAME, getAppName } from './config-params';
+import { store } from '../redux/store';
+import ReconnectingWebSocket, { Event } from 'reconnecting-websocket';
+import { AppState } from '../redux/reducer';
+
+export interface ErrorWithStatus extends Error {
+    status?: number;
+}
+
+export type Url = Exclude<Parameters<typeof fetch>[0], Request>; //string | URL;
+export type InitRequest = Partial<Parameters<typeof fetch>[1]>; //Partial<RequestInit>;
+export type Token = string;
+export type ReqResponse = Awaited<ReturnType<typeof fetch>>;
```

Comment:
Return type of fetch is a Response object, it is a known API, why not using it ?